### PR TITLE
bpo-38316: describe co_stacksize a little bit better

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue38316

### CODE REVIEW

File: Doc/reference/datamodel.rst [hunk 1]
```diff
@@ -925,8 +925,9 @@ Internal types
       the first line number of the function; :attr:`co_lnotab` is a string
       encoding the mapping from bytecode offsets to line numbers (for details
       see the source code of the interpreter); :attr:`co_stacksize` is the
-      required stack size (including local variables); :attr:`co_flags` is an
-      integer encoding a number of flags for the interpreter.
+      largest size, stack will ever grow to (after certain operations);
```
Comment (ammaraskar) [inline Doc/reference/datamodel.rst]: Let's get rid of the comma here: "is the largest size the stack will ever grow to"

and maybe reword "after certain operations" to "due to certain instructions" or maybe even omit that part.

**Comment by ammaraskar** [comment]
hmm, now that I think about this more. I think more appropriate wording would be:

> `co_stacksize` is the largest size the stack will grow to when executing the function;

**Comment by isidentical** [comment]
@ammaraskar can you check it now?

File: Doc/reference/datamodel.rst [hunk 2]
```diff
@@ -925,8 +925,9 @@ Internal types
       the first line number of the function; :attr:`co_lnotab` is a string
       encoding the mapping from bytecode offsets to line numbers (for details
       see the source code of the interpreter); :attr:`co_stacksize` is the
-      required stack size (including local variables); :attr:`co_flags` is an
-      integer encoding a number of flags for the interpreter.
+      largest size stack will ever grow to when executing the function;
```
Comment (ammaraskar) [inline Doc/reference/datamodel.rst]:
```suggestion
      largest size the stack will ever grow to when executing the function;
```

**Review by ammaraskar — APPROVED** [review]
LGTM with final suggestion

**Comment by ammaraskar** [comment]
@vstinner Would you mind taking a look at this?

**Review by vstinner — CHANGES_REQUESTED** [review]
Objects/frameobject.c allocates an array of `code->co_stacksize + code->co_nlocals + ncells + nfrees` items. The array size doesn't chagne during the execution of the function, it's not like x86 machine mode which manipulates the stack pointer (SP) register and use PUSH/POP.

> :attr:`co_stacksize` is the required stack size (including local variables)

So this definition sounds wrong to me.

Instead, I suggest this definition:

```
:attr:`co_stacksize` is the required stack size
```

=> just remove `(including local variables)`.

**Comment by isidentical** [comment]
I have made the requested changes; please review again

**Comment by vstinner** [comment]
@isidentical: The automated backport to 3.7 and 3.8 failed. Can you please try to manually backport these changes? Use " cherry_picker d587272fe3b0fcad2f23a490e76f9f82ca7d64ef 3.8" or "git cherry-pick -x d587272fe3b0fcad2f23a490e76f9f82ca7d64ef".

**Comment by vstinner** [comment]
Ping @isidentical, tell me if you prefer that I do the backport.

**Comment by isidentical** [comment]
Thanks for the ping @vstinner, I totally forgot backports. Sending them right away
